### DISCUSSION TITLE
Example on how to use NonceProvider

### DISCUSSION THREAD

**Discussion by ianpogi5** [JedWatson/react-select#4734] [Q&A]
Any full example on how to use NonceProvider? 

I am trying to avoid using `unsafe-inline` in my project.

**Comment by Rall3n** [comment]
Hello @ianpogi5,

the `NonceProvider` component is just a wrapper around the `CacheProvider` component from [emotion](https://emotion.sh/docs/cache-provider#gatsby-focus-wrapper).

The best place for it to live would be around your app.
```jsx
<NonceProvider cacheKey="my-cache" nonce="my-nonce">
    <App />
</NonceProvider>
```

As it is a wrapper around `CacheProvider`, it will affect any other component using emotion.
